pyomo/contrib/pyros/util.py

```diff
@@ -280,8 +280,21 @@ def transform_to_standard_form(model):
     return
 
 
+def get_vars_from_objective(block):
```

Comment:
This is fine,  However, this method and `get_vars_from_constraints` are basically identical.  You might consider replacing both with a common method,
```python
def get_vars_from_component(block, ctype):
    """Determine all variables used in active components within a block.

    Parameters
    ----------
    block: Block
        The block to search for components.  This is a recursive
        generator and will descend into any active sub-Blocks as well.
    ctype:  class
        The component type (typically either :py:class:`Constraint` or
        :py:class:`Objective` to search for).

    """

    seen = set()
    for compdata in block.component_data_objects(
            ctype,
            descend_into=True,
            active=True):
        for var in EXPR.identify_variables(compdata.expr):
            if id(var) not in seen:
                seen.add(id(var))
                yield var
```

Alternatively, `component_data_objects()` accepts either a single ctype, or an iterable of ctypes to look for.  Seeing as you combine the results for objectives and constraints anyways, you could get it all in one pass with
```python
ComponentSet(get_vars_from_component(model, (Objective, Constraint))
```

Reply:
See subsequent commit.